tests/test.linter.js

```diff
@@ -183,6 +183,38 @@ describe('Linter', function() {
       });
   });
 
+  it('Eslint ignore patterns and .eslintignorerc should be ignored', () => {
+    // Verify https://github.com/mozilla/addons-linter/issues/1288 is fixed
+    var addonLinter = new Linter({_: [
+      'tests/fixtures/webextension_node_modules_bower']});
+
+    // Stub print to prevent output.
+    addonLinter.print = sinon.stub();
+
+    assert.equal(addonLinter.collector.warnings.length, 0);
+    assert.equal(addonLinter.collector.warnings.length, 0);
+
+    var getFileSpy = sinon.spy(addonLinter, 'scanFile');
+
+    return addonLinter.scan()
+      .then(() => {
+        // Ensure no warnings are raised. If we would not overwrite eslint
+        // ignore patterns eslint ignores `node_modules` and
+        // `bower_components` by default. We want them to be scanned though.
+        assert.equal(addonLinter.collector.warnings.length, 0);
+
+        // We have an expected SyntaxError in index.js, this verifies
+        // that .eslintignore is also ignored.
+        assert.equal(addonLinter.collector.errors.length, 1);
+
+        // Make sure we also scan them ourselves.
+        assert.ok(getFileSpy.calledWith('index.js'));
+        assert.ok(getFileSpy.calledWith('manifest.json'));
+        assert.ok(getFileSpy.calledWith('node_modules/foo.js'));
```

Comment:
But I still think it would be nice to have a file that causes a warning or error inside bower and node folders.